### PR TITLE
Introduce DoctrineORMProxyQueryInterface

### LINKED ISSUE #1216: This project is using methods which does not exist in ProxyQueryInterface without exposing a custom interface
Related to the discussion: https://github.com/sonata-project/SonataDoctrineORMAdminBundle/pull/1199#issuecomment-732333757

When methods have `ProxyQueryInterface` param, we're using them like it was always a `ProxyQuery`.
We should add a type-check before.

Currently the `ProxyQuery` is not final, so
```
$query instanceof ProxyQuery
```
Can do the job, but exposing an interface `DoctrineProxyQueryInterface extends ProxyQueryInterface` could be better in order to declare `ProxyQuery` as final and allow others implementation.

The main issue with the DoctrineProxyQueryInterface (which should basically be ProxyQueryInterface + QueryBuilder) are:
- There is no QueryBuilderInterface we can extends.
- If we declare our DoctrineProxyQueryInterface with all the methods from the QueryBuilder, we'll have to add them to our ProxyQuery since __call is not enough to implement an Interface.

### PR DESCRIPTION
**Pull request by VincentLanglet**
## Subject

I am targeting this branch, because BC.
Then I'll replace every magic `foo` call by `getQueryBuilder->foo()`

Closes #1216.

## Changelog

```markdown
### Added
- Added `Sonata\DoctrineORMAdminBundle\Datagrid\ProxyQueryInterface`
```

### CODE REVIEW

File: src/Datagrid/DoctrineORMProxyQueryInterface.php [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<?php
+
+namespace Sonata\DoctrineORMAdminBundle\Datagrid;
+
+use Doctrine\ORM\QueryBuilder;
+use Sonata\AdminBundle\Datagrid\ProxyQueryInterface;
+
+interface DoctrineORMProxyQueryInterface extends ProxyQueryInterface
```
Comment (phansys) [inline src/Datagrid/DoctrineORMProxyQueryInterface.php:8]:
```suggestion
use Sonata\AdminBundle\Datagrid\ProxyQueryInterface as BaseProxyQueryInterface;

interface ProxyQueryInterface extends BaseProxyQueryInterface
```

**Review by core23 — APPROVED** [review]

**Review by phansys — APPROVED** [review]

File: src/Datagrid/ProxyQueryInterface.php [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+namespace Sonata\DoctrineORMAdminBundle\Datagrid;
+
+use Doctrine\ORM\QueryBuilder;
+use Sonata\AdminBundle\Datagrid\ProxyQueryInterface as BaseProxyQueryInterface;
+
+interface ProxyQueryInterface extends BaseProxyQueryInterface
+{
+    /**
+     * @return QueryBuilder
```
Comment (phansys) [inline src/Datagrid/ProxyQueryInterface.php:22]: Should we add a `NEXT_MAJOR` comment indicating that this docblock must be moved to the return type declaration?

Comment (core23) [reply src/Datagrid/ProxyQueryInterface.php:22]: 👎 That should be clear. We need to do this for every method.

Comment (phansys) [reply src/Datagrid/ProxyQueryInterface.php:22]: Good. It's just the panic for the lack of the interface.

Comment (VincentLanglet) [reply src/Datagrid/ProxyQueryInterface.php:22]: https://github.com/sonata-project/SonataDoctrineORMAdminBundle/issues/1188

**Pull request merged**